src/common/formatter_filter.cc

```diff
@@ -0,0 +1,1583 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <boost/program_options/variables_map.hpp>
+#include <boost/program_options/parsers.hpp>
+
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include <time.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include "global/global_init.h"
+#include "common/ceph_argparse.h"
+#include "include/stringify.h"
+#include "common/errno.h"
+#include "common/safe_io.h"
+
+#include "os/bluestore/BlueFS.h"
+#include "os/bluestore/BlueStore.h"
+#include "common/admin_socket.h"
+#include "common/url_escape.h"
+
+#include "formatter_filter.h"
+/*
+  expression:= literal ||
```

Comment:
this handwriting tokenizer and parser are difficult to understand. i'd suggest rewrite it with `boost::spirit`. i can help with this part if needed.

Reply:
I like the idea.
If examples in crush parsing will not be enough, I will capitalize on your offer of help.